addons/disposable/functions/fnc_handlePreparation.sqf

```diff
@@ -0,0 +1,30 @@
+/*
+ * Author: Ledere
+ * Handle preparation animation for supported launchers.
+ *
+ * Arguments:
+ * 0: unit - Object the event handler is assigned to <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [fromWeaponEH] call ace_disposable_fnc_handlePreparation;
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_unit","_weapon"];
+
+//If it's an AI, if our current weapon is not a launcher, if our current launcher is not a hasPreparation launcher, or if our launcher is actually a used launcher, exit.
+if (!local _unit || !([_unit] call EFUNC(common,isPlayer)) || {_weapon != secondaryWeapon _unit} || {getNumber (configFile >> "CfgWeapons" >> _weapon >> QGVAR(hasPreparation)) != 1} || {getNumber (configFile >> "CfgWeapons" >> _weapon >> "ACE_isUsedLauncher") == 1})  exitWith {};
```

Comment:
Using a _config var would require evaluation of the configFile for every weapon that is passed through, and according to CBA's documentation the weaponEH is called everytime a weapon is switched. I'm fine with using a _config var, I just felt that this method would be faster in the long term.